### PR TITLE
Fixes #3465

### LINKED ISSUE #3465: Package.on_use(), api.add_files() does not handle css media queries correctly
As far as I understand

    Package.on_use(function (api, where) {
        api.add_files('styles/some.css', 'client');
    }

should translate relative paths to absolute paths when bundling packages. This works for "regular" css, but not for css inside media queries.

Works:

    .awesome-marker-soft {
        background: url('../images/markers-soft.png') no-repeat 0 0;
    }

(URL will be translated to '/packages/[packagename]/images/markers-soft.png')

Does not work:

    @media (min--moz-device-pixel-ratio: 1.5),(-o-min-device-pixel-ratio: 3/2),(-webkit-min-device-pixel-ratio: 1.5),(min-device-pixel-ratio: 1.5),(min-resolution: 1.5dppx) {
        .awesome-marker {
            background-image: url('../images/markers-soft@2x.png');
        }
    }

(URL will stay '../images/markers-soft@2x.png')

To reproduce:

    meteor create reproduce
    cd reproduce
    git init
    git submodule add https://github.com/richsilv/meteor-leaflet-awesome-markers.git packages/richsilv:leaflet-awesome-markers
    meteor add richsilv:leaflet-awesome-markers
    meteor

### PR DESCRIPTION
**Pull request by RobertLowe**
Fix all the things! \o/

In the `ast` rules with a `type` of `media` have nested `rules`. We now recurse anytime we notice nested `rules`.

YAY

### CODE REVIEW

**Comment by Slava** [comment]
Yay! Thanks @RobertLowe!

I am going to review it later during the week. In the meanwhile, can you please add a test for this particular fix?

**Comment by RobertLowe** [comment]
@Slava DONE! :+1:

File: packages/minifiers/minifiers.js [hunk 1]
```diff
@@ -88,48 +88,59 @@ CssTools = {
...
-          if (isRelative(resource.path) && resource.protocol === null) {
-            absolutePath = pathJoin(basePath, resource.path);
-            newCssUrl = "url(" + quotes + absolutePath + quotes + ")";
-            value = value.replace(oldCssUrl, newCssUrl);
-          }
+    var rewriteRules = function (rules) {
+      _.each(rules, function(rule, ruleIndex) {
+
+        // do we need to recurse?
+        if (rule.rules != null) {
```
Comment (Slava) [inline packages/minifiers/minifiers.js]: What does this check mean? Can we make it more specific? The comment above isn't very descriptive either.

**Comment by Slava** [comment]
Merged with minor changes. Thank you.

**Comment by RobertLowe** [comment]
@Slava thanks I'll use that style next time. :+1:

**Pull request closed without merging**